giraffe/src/types/index.ts

```diff
@@ -494,6 +508,18 @@ export enum SpecTypes {
   Table = 'table',
 }
 
+export interface AnnotationLayerSpec {
+  type: 'annotation'
+  table: Table
+  annotations: AnnotationMark[] // annotations that are renderable (within the domain)
```

Comment:
what about calling this something along the lines of `visibleAnnotations` | `renderableAnnotations` since that's what they're representing, right?

Reply:
That's a good point, `annotations` can be confusing because both the config and the spec have that property. Let's use `annotationsData` to be a bit more generic than "visible" or "renderable". The spec is just a place to store valid data.  And this matches up well with LineLayerSpec using "lineData" as a property.